editor/js/reaction.js

```diff
@@ -198,16 +365,58 @@ function addSlowly(template, root) {
   node.show('slow');
   dirty();
   listen(node);
+  $('[data-toggle="tooltip"]', node).tooltip();
   return node;
 }
 
 // Removes from the DOM the nearest ancestor element matching the pattern.
 function removeSlowly(button, pattern) {
   const node = $(button).closest(pattern);
-  node.hide('slow', () => node.remove());
+  // Must call necessary validators only after the node is removed,
+  // but we can only figure out which validators these are before removal.
+  // We do so, and after removal, click the corresponding buttons to trigger validation.
+  let buttonsToClick = $();
```

Comment:
Yes, validate after removal. Is it okay if validation fails, because removal's already been done?

Reply:
The question is whether removing can cause the result to be invalid.

If so, then I think either there should be an error message, even if there is no way to go back, so the person can know their data have become corrupted.

If not, then I don't see why we can't just skip the validation and simplify the code.

Reply:
Realized I might've been misunderstanding -- just to check, "invalid" as in "fails validation checks"? Not that validation itself goes awry?

If so -- yes, removal can cause an invalid result, since the messages it's a part of can become invalid. This would be noted in the messages' indicators (the error counts would increase). Would this be enough?

Reply:
So long as the user can see they created a problem, I don't have a better idea for how to recover this besides creating an undo stack.

I do think we should create an undo stack.

Reply:
(Not in this PR.)